ops/pebble.py

```diff
@@ -1464,45 +1465,28 @@ def pull(self, path: str, *, encoding: str = 'utf-8') -> typing.Union[typing.Bin
         if not boundary:
             raise ProtocolError('invalid boundary {!r}'.format(boundary))
 
-        # We have to manually write the Content-Type with boundary, because
-        # email.parser expects the entire multipart message with headers.
-        parser = email.parser.BytesFeedParser()
-        parser.feed(b'Content-Type: multipart/form-data; boundary='
-                    + boundary.encode('utf-8') + b'\r\n\r\n')
-
-        # Then read the rest of the response and feed it to the parser.
+        parser = MultipartLargeFileParser(boundary)
         while True:
             chunk = response.read(8192)
             if not chunk:
                 break
             parser.feed(chunk)
-        message = parser.close()
-
-        # Walk over the multipart parts and read content and metadata.
-        resp = None
-        content = None
-        for part in message.walk():
-            name = part.get_param('name', header='Content-Disposition')
-            if name == 'response':
-                resp = _json_loads(part.get_payload())
-            elif name == 'files':
-                filename = part.get_filename()
-                if filename != path:
-                    raise ProtocolError('path not expected: {}'.format(filename))
-                # decode=True, ironically, avoids decoding bytes to str
-                content = part.get_payload(decode=True)
 
+        resp = parser.get_response()
         if resp is None:
             raise ProtocolError('no "response" field in multipart body')
         self._raise_on_path_error(resp, path)
 
-        if content is None:
+        filenames = parser.filenames()
+        if len(filenames) < 1:
             raise ProtocolError('no file content in multipart response')
-        if encoding is not None:
-            reader = io.StringIO(content.decode(encoding))
-        else:
-            reader = io.BytesIO(content)
-        return reader
+        if len(filenames) > 1:
+            raise NotImplementedError('Multiple file responses not yet supported')
```

Comment:
I wonder if this error could be a tiny bit clearer. You just did a 'pull(path)' there really shouldn't be any way to get >1 filename back, right?

Reply:
In addition to the comments from John, please note that the error messages are consistently lowercased, so that the typical "joining: like: this" works well visually.  Log messages are consistently titled-cased, though, for similar reasons (they are typically full sentences in the context in which they are presented).

Reply:
With the caveat that I haven't gone in to review what would happen right now if you threw something like `/home/ubuntu/somefiles/*.txt` at the pebble API, I kind of like this error.

In most cases, we will never reach this branch in the code. But one of the edge cases where we do hit it is when (and if) the Pebble API is updated to support a star or some other method of specifying multiple files in the path, and someone tries to be clever and just pass that through this function. In that case, "NotYetImplemented" is definitely the correct error to raise, as we do want to implement multi file support in the future. We just haven't done so yet!

(That said, we should lowercase the M)

Reply:
lower cased